### PR TITLE
Persist API: more logging and fix tracing error

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes
Adding logging when call to persist API fails
Fix tracing error in persist API

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/persist/lib/controllers/persist.controller.ts [hunk 1]
```diff
@@ -270,16 +270,16 @@ class PersistController {
                 content: `There was an issue with the batch ${persistType}. ${error?.message}`,
                 timestamp: Date.now()
             });
-            const errMsg = `Failed to ${persistType} records ${activityLogId}`;
-            span.setTag('error', errMsg).finish();
-            return err(errMsg);
+            const res = err(`Failed to ${persistType} records ${activityLogId}`);
+            span.setTag('error', res.error).finish();
```
Comment (TBonnin) [inline packages/persist/lib/controllers/persist.controller.ts:274]: `tracer.setTag('error', e)` e must be an error, not a string. see https://docs.datadoghq.com/tracing/trace_collection/custom_instrumentation/dd_libraries/nodejs/?tab=errors

File: packages/shared/lib/sdk/sync.ts [hunk 2]
```diff
@@ -570,7 +571,14 @@ export class NangoSync extends NangoAction {
                     lastSyncDate: date
                 }
             });
-            return response.status <= 299;
+            if (response.status > 299) {
+                console.log(
+                    `Request to persist API (setLastSyncDate) failed: errorCode=${response.status} response='${JSON.stringify(response.data)}'`,
+                    JSON.stringify(this)
+                );
+                throw new Error(`cannot set lastSyncDate for sync '${this.syncId}'`);
```
Comment (TBonnin) [inline packages/shared/lib/sdk/sync.ts:592]: operation results are rarely (or even ever) checked in integrations. Let's throw an error to make it fails loudly instead

Comment (khaliqgant) [reply packages/shared/lib/sdk/sync.ts:592]: How does this error surface to the end user?

**Review by khaliqgant — APPROVED** [review]
Question inline

File: packages/shared/lib/sdk/sync.ts (hunk 2 above)
Comment (TBonnin) [reply packages/shared/lib/sdk/sync.ts:592]: if it is not catched and handled by the integration, it is captured as a failure of the sync
Screenshot 2024-02-01 at 07 57 11

**Pull request merged**